From 3103d307063c71ff7791c2a59aacd41cfa05285c Mon Sep 17 00:00:00 2001
Message-Id: <3103d307063c71ff7791c2a59aacd41cfa05285c.1676544357.git.pmatilai@redhat.com>
From: Panu Matilainen <pmatilai@redhat.com>
Date: Thu, 16 Feb 2023 12:18:02 +0200
Subject: [PATCH 1/2] Exclude kernel modules from ELF dependency generation

The kernel modules are technically ELF DSO's but we only care about
library level dependencies which the kernel modules will never have, so
processing them is just waste of time.

This "regressed" when we dropped the "exeonly" flag from ELF in
commit 8901a6be1652519c4875436e29e0a663be0790b7.
---
 fileattrs/elf.attr | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fileattrs/elf.attr b/fileattrs/elf.attr
index 58322d3cd..21bc471b0 100644
--- a/fileattrs/elf.attr
+++ b/fileattrs/elf.attr
@@ -1,3 +1,4 @@
 %__elf_provides		%{_rpmconfigdir}/elfdeps --provides
 %__elf_requires		%{_rpmconfigdir}/elfdeps --requires
 %__elf_magic		^(setuid,? )?(setgid,? )?(sticky )?ELF (32|64)-bit.*$
+%__elf_exclude_path	^/lib/modules/.*\.ko?(\.[[:alnum:]]*)$
-- 
2.39.1

